src/generator/state-machine-to-intermediate.js

```diff
@@ -154,18 +154,50 @@ const getIntermediateFormOf = ({left, right, operator, out}, isSecondarySignal)
     }
 };
 
+// Helper function to generate next available numeric
+let numericStateEnd;
+function generateNextLabel(state, states) {
+    if (numericStateEnd === undefined) {
+        numericStateEnd = endOfLastNumericState(states);
+    }
+
+    const ends = state.statements.map(s => s.start + s.operations.length);
+    let end = getMaxRounded(ends);
+    let tmp = numericStateEnd;
+    numericStateEnd = numericStateEnd + end;
+    return tmp;
+}
+
+function getMaxRounded(ends) {
+    let end = Math.max(...ends) + 1;
+    end = Math.max(end, 0); // prevent -infinity when ends is empty
+    return Math.ceil(end / 10) * 10; // Round to the nex multiple of 10
+}
+
+function endOfLastNumericState(states) {
+    const ends = states
+        .filter(({state}) => typeof state === 'number')
+        .map(({statements}) => statements.map(s => s.start + s.operations.length))
+        .flat();
+    return getMaxRounded(ends);
```

Comment:
This all looks really good. One final suggestion:

```suggestion
    return Math.max(0, ...ends) + 1;
```

I don't think it's necessary to threshold the label to 10 since the user never sees the generated state numbers. Taking it out would remove the need for the helper function.

Unless.... we wanted to make _every_ generated state be thresholded so that IDEs could give the information in tooltips using static code analysis. Would this be a useful feature? Because right now, we're just thresholding the first string-labelled state which doesn't make a lot of sense.

Reply:
The code was rounding all the generated labels to the next multiple of 10 (10, 20, ...) but yeah, this is not strictly necessary, so I removed it